### PR TITLE
CSPL-1823: Add utils to support the use of Azure Blob Storage for App Framework and Smartstore tests

### PR DESCRIPTION
**Pull request by jambrosiano**

### CODE REVIEW

File: test/env.sh [hunk 1]
```diff
@@ -30,6 +39,8 @@
 : "${SKIP_REGEX:=}"
 # Set to DEBUG_RUN:=True to skip tear down of test environment in case of test failure
 : "${DEBUG_RUN:=False}"
+# Type of deployment, manifest files or helm chart, possible values "manifest" or "helm"
+: "${DEPLOYMENT_TYPE:=manifest}"
```
Comment (gaurav-splunk) [inline test/env.sh:43]: are we hardcoding it here or should we leave it blank?

File: test/testenv/azureutils.go [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+	count := 0
+	max := len(dataWithKey.Blobs.Blob)
+
+	for count = 0; count < max; count++ {
+		blob := dataWithKey.Blobs.Blob[count]
+		fmt.Println("Azure blob app details", "Count:", count, "App name:", blob.Name, "Etag:", blob.Properties.ETag,
+			"Created on:", blob.Properties.CreationTime, "Modified on:", blob.Properties.LastModified)
+	}
+
+	fmt.Println("Azure blob with secret key done")
```
Comment (gaurav-splunk) [inline test/testenv/azureutils.go:261]: what does this msg mean?

**Comment by kumarajeet** [comment]
Approving with small note:
@jambrosiano  and I will revisit the refactoring comments I left in https://github.com/splunk/splunk-operator/pull/884 in one of the upcoming sprints.

File: test/testenv/azureutils.go [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+		return false
+	}
+	fmt.Println("Azure blob app download", "Resp Status", respAppDownload.Status)
+	fmt.Println("Azure blob app download", "Resp Status", respAppDownload.Status)
+	fmt.Println("Azure blob app download", "Contenth Length", respAppDownload.ContentLength)
+	return true
+}
+
+// UploadFileFromAzure uploads a file to Azure Storage account container
+func UploadFileFromAzure(accountName, accountKey, appPackageFullPath, localFileName string) bool {
```
Comment (kumarajeet) [inline test/testenv/azureutils.go:90]: nit: UploadFileToAzure

**Review by kumarajeet — APPROVED** [review]

**Comment by tpereirasplunk** [comment]
Should we replace the `fmt.Println` with `logf.Log.`

**Review by tpereirasplunk — APPROVED** [review]

**Review by gaurav-splunk — APPROVED** [review]

**Pull request merged**